### PR TITLE
Use `ABORTED` from `Run.reload`

### PR DESCRIPTION
**Pull request by jglick**
Resolving a comment I made a decade ago (!) in b55c157813c0f5dc3cb06d0503834d21115aebc9: if a non-Pipeline build is interrupted by a controller restart, `ABORTED` status (used for all sorts of sudden interruptions unrelated to the logical build steps) is more natural than `FAILURE` (a problem in a specific step preventing the build from proceeding). This is also more consistent with Pipeline builds, or `node` blocks, that fail to resume for various reasons when normally they could.

### Testing done

None; apparently this does not have test coverage? Could add some if it seems valuable.

### Proposed changelog entries

- Non-Pipeline builds interrupted by a controller restart will now be marked as aborted rather than failed.

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by Vlatombe — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]
I like that change, I was more than once asked why it says failed instead of aborted in case of controller restarts.

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
